### PR TITLE
fix reading of notifications and bump dependencies

### LINKED ISSUE #2: Error when using notifications
I've an issue on Ubuntu 22.04, where using subscriptions would result in an application crash.

The error returned is
```
Error: BtleError(Other(DbusError(D-Bus error: Read not permitted (org.bluez.Error.NotPermitted))))
``` 

## Behavior
Everything works as expected (including read) until a notification / event is received. Then the call to `read_deserialize` in `listen_push_events` will fail with the above mentioned error. I've been able to reliably produce this with either attaching/disconnecting the cup from a charging base or by temperature changes.

## Reproducing it
I've been able to reproduce this error with the provided egui application and I can confirm that it works with e.g. the python ember cup library.

This is a minimal (non-working) example:
```rust
use std::error::Error;
use tokio;
use ember_mug::EmberMug;
use futures_util::{StreamExt, TryStreamExt};

mod backend;

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let mug = EmberMug::find_and_connect().await?;

    mug.subscribe_push_events().await?;
    let mut stream = mug.listen_push_events().await?.boxed();

    // everything up here is fine, until a state change occurs
    while let Some(event) = stream.try_next().await? {
        println!("Received push");  // it will never reach this point
    }

    Ok(())
}
```

## Ideas
I currently expect this to be a general error with the bluetooth library stack used here, but I'm unsure how to debug this as I'm fairly new to rust.

### PR DESCRIPTION
**Pull request by Emilgardis**
fixes #2

### CODE REVIEW

**Comment by tyalie** [comment]
Can confirm that it works

File: src/mug/push_events.rs [hunk 1]
```diff
@@ -33,10 +33,7 @@ impl EmberMug {
             .await?
             .filter_map(move |v| async move {
                 if crate::KnownCharacteristic::PushEvents.get() == v.uuid {
-                    Some(
-                        self.read_deserialize(&crate::KnownCharacteristic::PushEvents)
-                            .await,
-                    )
+                    Some(PushEvent::read(&mut Cursor::new(v.value)).map_err(ReadError::ReadError))
```
Comment (Emilgardis) [inline src/mug/push_events.rs:36]: I'm surprised this worked, oh well, thanks again @tyalie :D

**Comment by tyalie** [comment]
Oh i forgot to mention: I think `get_push_event` isn't really fixable. I thought about adding a debug statement there that mentions that this prob. doesn't work on some systems - which would keep compatibility, but makes note of the issue if anybody hits upon it again.

**Comment by Emilgardis** [comment]
i don't know what's wrong with nightly, getting 

```
  --> src/mug/battery.rs:12:10
   |
12 | #[derive(BinRead, Debug)]
   |          ^^^^^^^
   |
   = help: message: attempt to subtract with overflow
```

File: src/mug/mug_meta.rs [hunk 2]
```diff
@@ -2,19 +2,37 @@ use super::*;
...
+    Vec<u8>: FromIterator<T>,
+    Reader: std::io::Read + std::io::Seek,
+    Arg: Clone,
+{
+    let res: Vec<u8> = binrw::helpers::until_eof::<Reader, T, Arg, Vec<u8>>(reader, endian, args)?;
+    String::from_utf8(res).map_err(|e| binrw::Error::Custom {
+        pos: 6 + 1,
+        err: Box::new(e) as _,
+    })
+}
```
Comment (Emilgardis) [inline src/mug/mug_meta.rs:38]: https://github.com/jam1garner/binrw/issues/239

**Pull request merged**